### ISSUE TITLE
Implement bash autocompletion (for dosbox command)

### ISSUE THREAD

**Issue by dreamer** [dosbox-staging/dosbox-staging#400]
We need normal bash autocompletion script to be installed alongside dosbox to help users with filling out `dosbox` parameters (e.g. `-conf`); basically all the options listed in our manual.

**Comment by FeignInt** [comment]
Hi to all

I've started one

it is missing

- -c 
- -socket
- -lang 
They can be selected but don't trigger anything

editconf and opencapture  only have a basic selection , vim or nano or whatever is in $EDITOR
and for media, $BROWSER , firefox and pcmanfm.. 
just need to populate their arrays. It checks availability using `type -P` output is a little ugly as I've not trimmed path off them

The -conf will look in `XDG_CONFIG_HOME/dosbox` and `PWD` for `conf|CONF` the available list shrinks as they are selected.

The none launching options , things like -version, printfconf are only available as first option, and prevent all other options, besides the selection of editor/browser

single use options are hidden once used
-c and -conf can be reused
I have just realised that I neglected to turn on `-userconf` :(


It will let you navigate filesystem, but It is not feature rich, i.e. no exe|com|bat filtering

The function(s) that deal with the hiding of options are awful, I want to re-write those.


[it can be found here](https://github.com/FeignInt/dosbox-staging-deb/tree/dosbox_bash-completion)

just source the file /!\ read it first /!\
````
source dosbox_completion
. dosbox_completion #  that is  [dot|fullstop|period] dosb... 
```

I would suggest an alias while testing
`alias dosbox=echo`
and you can either unset it, or ignore it 

```
unset dosbox # removes alias
\dosbox          # will execute the real one, leaving alias
``` 

 let me know what editors you all use
how do you organise your .confs?

**Comment by dreamer** [comment]
Haven't tested it yet, sounds great :)

Few notes:

- `-editconf` option: as of 0.76.0 (next release), dosbox-staging "guesses" the editor program on its own, so you don't need to (and yes, it also looks into EDITOR variable). Also, the same option in GNU style was added (`--editconf`). Therefore, we would prefer to promote user running: `dosbox --editconf` instead of old variant (`dosbox -editconf vim`). (old variant still works, but I would prefer to remove it at some point)
- Make `--editconf` available as first option only, please.
- `-printconf` now exists as `--printconf`, and we would prefer users to switch to GNU style (old variant still works for compatibility) (since 0.76.0)
- `--version` exists as replacement for `-version` (since 0.75.0)
- I am ok with neglecting `-userconf` for now - I think it does not work as it is supposed to; we can add it once it's fixed (I am not even sure *how it is supposed to work at this point*)

As for selection of file for launching, we want to auto-fill the following ones:
- directories (passing directory mounts it as C drive)
- `.exe`, `.bat` and `.com` files

If you plan to send it as PR (PR will trigger shellcheck on the script), then we want this script in `contrib/linux/` (maybe `contrib/linux/bash-completion/`?).

Also, are you working on dosbox-staging Debian package? Because that would be awesome :)

**Comment by FeignInt** [comment]
- long options should be no problem

yeah, I'll rename to 
```
contrib/linux/bash-completion/dosbox
```
 before integrating into main tree

regards Debian packaging, I have not done anything for a while
I have been able to compile for sid, bullseye, buster and Ubuntu Bionic.
The latter required a small tweak as bionc has a slightly older version of debhelper. I've not looked properly at the difference
it makes. bionic's version 11 is actively discouraged on Sid
However, the issue seems to be related to systemd services, which is not used, so it is probably fine to simple set to compat11 instead of 12 
I'll see if I can take a proper look at it again over the weekend.
maybe push an apt-repo for some test packages.
I've only tested installs on sid,
  
one thing that was worrying me was the copyright file
I have since found "debmake" which can generate a machine readable copyright file from the source files, so I will play around with that

if anyone is interested in building/testing on debian
https://github.com/FeignInt/dosbox-staging-deb

none Sid<unstable> and Ubuntu users are especially welcome

compiling not your thing?
no problem
cleaning up the wordy bits like [the description in control](https://github.com/FeignInt/dosbox-staging-deb/blob/debianish/debian/control) is needed

I don't know how much time I could put in to maintaining it
so if it were adopted it would make sense to set the Maintainer:  to something like "dosbox-staging team <email@address.org>
